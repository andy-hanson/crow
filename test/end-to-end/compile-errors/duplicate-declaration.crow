main void()
	()

foo record
	a nat
	a nat

foo union
	b
	b nat

foo enum
	c
	c

foo flags
	d
	d

s spec
s spec

f nat(a nat, a nat)
	a

f nat()
	a = 1
	a = 2
	a

f nat(a nat)
	a = 1
	a

f nat()
	1 call a =>
		a = 1
		a

f nat()
	1 call a =>
		b = 1
		1 call b =>
			a = 2
			b

f nat()
	1 call a =>
		1 call a =>
			a

call nat(_ nat, f fun nat(nat))
	f[0]
