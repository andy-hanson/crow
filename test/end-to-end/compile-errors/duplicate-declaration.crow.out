exit code: 0
stderr:
test/end-to-end/compile-errors/duplicate-declaration.crow 6:5-6:10 Record field name 'a' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 8:1-8:4 Type name 'foo' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 10:5-10:10 Union member name 'b' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 12:1-12:4 Type name 'foo' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 14:5-14:6 Enum member name 'c' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 16:1-16:4 Type name 'foo' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 18:5-18:6 Flags member name 'd' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 21:1-21:2 Spec name 's' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 23:14-23:15 Local name 'a' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 28:5-28:6 Local name 'a' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 32:5-32:6 Local name 'a' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 37:9-37:10 Local name 'a' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 43:16-43:17 Local name 'b' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 44:13-44:14 Local name 'a' is already used.
test/end-to-end/compile-errors/duplicate-declaration.crow 49:16-49:17 Local name 'a' is already used.
